### ISSUE TITLE
YAML field order is sorted alphabetically for (at least) gfm output

### ISSUE THREAD

**Issue by florisvdh** [jgm/pandoc#7891]
**Problem:**

Using an [example `test.md`](https://github.com/jgm/pandoc/files/7996464/test.md) file (contents printed below), when requesting conversion to **gfm**, Pandoc sorts the YAML field alphabetically. The result is less accessible to a human, e.g. the title field is put at the bottom:

```bash
$ cat test.md 
---
title: "Example title"
description: "Example description"
date: 2022-01-31
output: 
  md_document:
    variant: gfm
---
    
Some text.
$ 
$ 
$ ~/apps/pandoc-2.17.1.1/bin/pandoc test.md -s -f markdown -t gfm
---
date: 2022-01-31
description: Example description
output:
  md_document:
    variant: gfm
title: Example title
---

Some text.
```

Is it possible to leave the order of the YAML fields unchanged?

The usecase where I encountered this is using the `rmarkdown` package from R to generate **gfm** output from Pandoc's markdown.

**Pandoc version:**

```
pandoc 2.17.1.1
Compiled with pandoc-types 1.22.1, texmath 0.12.4, skylighting 0.12.2,
citeproc 0.6.0.1, ipynb 0.2
```

**Comment by jgm** [comment]
It isn't currently possible, because the metadata is stored is a Data.Map.Map (hashmap), which doesn't preserve order.

**Comment by cderv** [comment]
FWIW, preserving the order is important for us so we do the following: 
* deactivate the extension for markdown writer, 
* convert without YAML metadata being kept 
* do a post processing to add the YAML header in the md output by Pandoc, the same as in the input. 

Feel free to close if this is something that is unlikely to change.

**Comment by sjakobi** [comment]
FYI, the package [`insert-ordered-containers`](https://hackage-content.haskell.org/package/insert-ordered-containers) offers a similar API as `Data.Map` but remembers the insertion order.

**Comment by jgm** [comment]
Yes, I know there are ordered map packages.  But moving to that would be an API change in pandoc-types, which I'd rather avoid...
